### DISCUSSION TITLE
A simple way to assign words to function keys.

### DISCUSSION THREAD

**Discussion by Whammo** [jkotlinski/durexforth#594] [Ideas]
There was a similar issue with control keys.
 
 ```forth

: 133 literal dos ;

```

**Comment by jkotlinski** [comment]
Is this a feature request?

**Comment by Whammo** [comment]
I guess it is a feature request. Or a request for insight as to how such a thing may be accomplished.

**Comment by jkotlinski** [comment]
![vice-screen-2025010112441371](https://github.com/user-attachments/assets/15f96084-3cce-4ea0-a542-28f90992c5ec)

This would be a way to do it, not sure if it fits your needs!

**Comment by Whammo** [comment]
I enjoyed figuring that out. Thanks for your effort.
What I was thinking was substituting a string for the F-key in regular input so if you're in the compiler, or in V the string is output, and the F-key is dropped.
 I thought of intercepting the IRQ vector, not simple.

**Comment by jkotlinski** [comment]
Ok… I do not think there is a single good place to do such a key-to-text
expansion.

For the interpreter, one would probably modify REFILL to add custom actions
to the Function keys.
But it would not carry over to v editor.

tors 2 jan. 2025 kl. 13:31 skrev Whammo ***@***.***>:
